### ISSUE TITLE
[BUG] Character substitution in names imported from NEXUS

### ISSUE THREAD

**Issue by SynedraAcus** [Cibiv/IQ-TREE#54]
Substitution of unacceptable characters in sequence IDs never happens if names are imported from NEXUS (as opposed to IDs in a FASTA file).

I have a bunch of complex names, something like this:

```
Asterionellopsis_glacialis,_Strain_CCMP134|CAMPEP_0199907168
Phatr2|1719
```

They have commas and pipes and whatnot. IQ-TREE itself accepts them just fine by substituting all offending chars with the underscore, but when I try to pass them through NEXUS file as cluster members for the likelihood mapping, it leaves a lengthy log like this:

```
Cluster 2 "diatom" lists 341 sequences:
WARNING: sequence name "Asterionellopsis_glacialis"! Will be ignored.
WARNING: sequence name ","! Will be ignored.
WARNING: sequence name "_Strain_CCMP134|CAMPEP_0199907168"! Will be ignored.
WARNING: sequence name "Phatr2|1719"! Will be ignored.
```

And so on for the rest of sequences. Of course, analysis either doesn't happen or proceeds with incorrect clustering. Running something like `s/[,|-]/_/g' on NEXUS file lets it proceed correctly. So the source is obvious: a NEXUS parser makes different assumptions about what a seq ID may and may not be.

**Comment by bqminh** [comment]
Thanks for the info. Right now we can't change the nexus parser as we use the nexus class library (NCL) and these special characters are delimiters for nexus file. Since there is a solution by pre-processing the nexus file as you suggested, I will mark this problem as low-priority, because we have other pressing issues to work on first.
Thank you, Minh

**Comment by bqminh** [comment]
This issue is likely fixed in v1.6.5. Can you please double check?
Thank you,.
